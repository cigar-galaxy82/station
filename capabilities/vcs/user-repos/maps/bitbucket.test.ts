import { SuperfaceTest } from '@superfaceai/testing-lib';

describe(`vcs/user-repos/bitbucket`, () => {
  let superface: SuperfaceTest;

  beforeEach(() => {
    superface = new SuperfaceTest();
  });

  describe('UserRepos', () => {
    it('should perform successfully - user with single repository', async () => {
      await expect(
        superface.run({
          profile: 'vcs/user-repos',
          provider: 'bitbucket',
          useCase: 'UserRepos',
          input: {
            user: 'jakuvacek',
          },
        })
      ).resolves.toMatchSnapshot();
    });
    it('should perform successfully - user with mutiple repositories', async () => {
      await expect(
        superface.run({
          profile: 'vcs/user-repos',
          provider: 'bitbucket',
          useCase: 'UserRepos',
          input: {
            user: 'JakubVacek',
          },
        })
      ).resolves.toMatchSnapshot();
    });
    it('should map error successfully', async () => {
      await expect(
        superface.run({
          profile: 'vcs/user-repos',
          provider: 'bitbucket',
          useCase: 'UserRepos',
          input: {
            user: '!!',
          },
        })
      ).resolves.toMatchSnapshot();
    });
  });
});
