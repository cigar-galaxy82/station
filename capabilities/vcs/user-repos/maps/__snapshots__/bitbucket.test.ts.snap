// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vcs/user-repos/bitbucket UserRepos should map error successfully 1`] = `
Err {
  "error": "MappedError: Expected error
Properites: {
  \\"message\\": \\"No workspace with identifier '!!'.\\",
  \\"statusCode\\": 404
}
AST Path: definitions[0].statements[0].statements[0]
Original Map Location: Line 11, column 5",
}
`;

exports[`vcs/user-repos/bitbucket UserRepos should perform successfully 1`] = `
Ok {
  "value": Object {
    "repos": Array [
      Object {
        "description": undefined,
        "name": "testRepository",
      },
    ],
  },
}
`;
